### PR TITLE
Feat/m register db instance

### PR DESCRIPTION
**Pull request by Abhishekism9450**
Register DB instance .

### CODE REVIEW

File: client/era/era_service.go [hunk 1]
```diff
@@ -267,3 +269,24 @@ func (sc ServiceClient) ListDatabaseInstance(ctx context.Context) (*ListDatabase
...
+	res := new(ProvisionDatabaseResponse)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return res, sc.c.Do(ctx, httpReq, res)
+}
+
+func (sc ServiceClient) UnRegisterDatabase(ctx context.Context, id string, req *UnRegisterDatabaseRequest) (*ProvisionDatabaseResponse, error) {
```
Comment (bhati-pradeep) [inline client/era/era_service.go]: Unregister is just delete with different input arguments right ?

File: nutanix/resource_nutanix_ndb_register_database.go [hunk 2]
```diff
@@ -0,0 +1,697 @@
...
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	era "github.com/terraform-providers/terraform-provider-nutanix/client/era"
+	"github.com/terraform-providers/terraform-provider-nutanix/utils"
+)
+
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_ndb_register_database.go:13]: I see computed values similar to that of create database resource, scaling, restore, etc. computed values. Here we can do this 
```suggestion
func resourceNutanixNDBRegisterDatabase() *schema.Resource {

        schema = &schema.Resource{
		CreateContext: resourceNutanixNDBRegisterDatabaseCreate,
		ReadContext:   resourceNutanixNDBRegisterDatabaseRead, 
		.
		.
		.
        // add computed values to existing schema
	schema.Schema.update(getComputedDatabaseShchema())
	return schema
		                        
```

Comment (bhati-pradeep) [reply nutanix/resource_nutanix_ndb_register_database.go:13]: ".update()" is just for logical explanation.

File: nutanix/resource_nutanix_ndb_register_database.go [hunk 3]
```diff
@@ -0,0 +1,697 @@
...
+	req, err := buildReisterDBRequest(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp, er := conn.Service.RegisterDatabase(ctx, req)
+	if er != nil {
+		return diag.FromErr(er)
+	}
+	log.Println(resp)
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_ndb_register_database.go]: Is this log required ?

File: nutanix/resource_nutanix_ndb_register_database.go [hunk 4]
```diff
@@ -0,0 +1,697 @@
...
+		Delay:   eraDelay,
+	}
+
+	if _, errWaitTask := stateConf.WaitForStateContext(ctx); errWaitTask != nil {
+		return diag.Errorf("error waiting for db register	 (%s) to create: %s", resp.Entityid, errWaitTask)
+	}
+	return nil
+}
+
+func resourceNutanixNDBRegisterDatabaseRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_ndb_register_database.go:311]: Database read can be used from database resource or dataasources.

File: nutanix/resource_nutanix_ndb_register_database.go [hunk 5]
```diff
@@ -0,0 +1,697 @@
...
+		}
+
+		if err := d.Set("linked_databases", flattenDBLinkedDbs(resp.Linkeddatabases)); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
+	return nil
+}
+func resourceNutanixNDBRegisterDatabaseUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_ndb_register_database.go]: Isn't it same as database resource's update ?

File: nutanix/resource_nutanix_ndb_register_database.go [hunk 6]
```diff
@@ -0,0 +1,697 @@
...
+		}
+
+		if err = d.Set("name", res.Name); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+	return nil
+}
+
+func resourceNutanixNDBRegisterDatabaseDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (bhati-pradeep) [inline nutanix/resource_nutanix_ndb_register_database.go:321]: Isn't it same as database resource delete ?

**Review by bhati-pradeep — COMMENTED** [review]
Many things are implemented in database resource. Please merge appropriate branches to import here.

File: nutanix/resource_nutanix_ndb_register_database.go (hunk 3 above)
Comment (Abhishekism9450) [reply nutanix/resource_nutanix_ndb_register_database.go]: no.  removed from code.

File: nutanix/resource_nutanix_ndb_register_database.go (hunk 5 above)
Comment (Abhishekism9450) [reply nutanix/resource_nutanix_ndb_register_database.go]: yes. called same create method from update.

File: client/era/era_service.go (hunk 1 above)
Comment (Abhishekism9450) [reply client/era/era_service.go]: yes. Removed this method .

File: nutanix/resource_nutanix_ndb_register_database.go (hunk 6 above)
Comment (Abhishekism9450) [reply nutanix/resource_nutanix_ndb_register_database.go:321]: yes. but with different inputs.

File: nutanix/resource_nutanix_ndb_register_database.go (hunk 4 above)
Comment (Abhishekism9450) [reply nutanix/resource_nutanix_ndb_register_database.go:311]: yes. now we are reading from database resource.

File: nutanix/resource_nutanix_ndb_register_database.go (hunk 2 above)
Comment (abhimutant) [reply nutanix/resource_nutanix_ndb_register_database.go:13]: will make the changes later as improvement.

**Review by siddharth-kulshrestha — APPROVED** [review]

**Review by abhimutant — APPROVED** [review]

**Pull request merged**